Dockerfiles/pcluster-amazonlinux-2/Dockerfile

```diff
@@ -128,6 +128,20 @@ RUN mkdir -p $(dirname "${SPACK_ROOT}") \
     && spack buildcache create -a -u /bootstrap-gcc-cache $(spack find --format '/{hash}') \
     && rm -rf $(dirname "${SPACK_ROOT}") /root/.spack
 
+# Sign the buildcache
+RUN --mount=type=secret,id=bootstrap_gcc_key \
+    mkdir -p $(dirname "${SPACK_ROOT}") \
+    && git clone https://github.com/spack/spack "${SPACK_ROOT}" \
+    && pushd "${SPACK_ROOT}" && git checkout ${SPACK_COMMIT} && popd \
+    && . "${SPACK_ROOT}/share/spack/setup-env.sh" \
+    && wc /run/secrets/bootstrap_gcc_key \
+    && spack gpg trust /run/secrets/bootstrap_gcc_key \
+    && secretkey_fingerprint=$(GNUPGHOME="${SPACK_ROOT}"/opt/spack/gpg gpg2 -K --with-fingerprint --with-colons | awk -F: '/fpr/{print $10}') \
+    && ls /bootstrap-gcc-cache/build_cache/*json | xargs -I {} spack gpg sign --output {}.sig --key ${secretkey_fingerprint} --clearsign {} \
+    && spack mirror add bootstrap-gcc-cache /bootstrap-gcc-cache \
+    && spack gpg publish --rebuild-index -m bootstrap-gcc-cache ${secretkey_fingerprint} \
```

Comment:
So, if I understand correctly here you are importing a secret key, which for the duration of the RUN step is on disk and is imported to Spack's `pubring.kbx`. The key is not persisted in the layer because of the last `rm` command.

Do we plan to rotate the key? Also, are these binaries meant to be used in production?

Reply:
> So, if I understand correctly here you are importing a secret key, which for the duration of the RUN step is on disk and is imported to Spack's pubring.kbx. The key is not persisted in the layer because of the last rm command.

Exactly.

> Do we plan to rotate the key? Also, are these binaries meant to be used in production?

The admins of this repo have the ability to rotate, but no plans to do so regularly at the moment. I left an explicit reference to the key name in the upload stage such that we get alerted when the key changes. This is to start the manual step involved after key changes, i.e. uploading to the sub-bucket created here: https://github.com/spack/spack-infrastructure/pull/731